src/cg.rs

```diff
@@ -35,22 +46,53 @@ impl CGImpl {
             view.addSubview_(subview); // retains subview (+1) = 2
             let _: () = msg_send![subview, release]; // releases subview (-1) = 1
         }
-        Ok(Self { layer, window })
+        let color_space = CGColorSpace::create_device_rgb();
+        Ok(Self {
+            layer,
+            window,
+            color_space,
+            width: 0,
+            height: 0,
+        })
     }
 
-    pub(crate) unsafe fn set_buffer(&mut self, buffer: &[u32], width: u16, height: u16) {
```

Comment:
It's less efficient, on backends that can avoid a copy. So the question is how often it is likely to be used in practice.

Reply:
It would be trivial to implement `set_buffer` on top of the no-copy API, though that would be less efficient on backends that can't do no-copy presentation. Which seems bad, so it would be a bit of code that has to be maintained in every backend and in `lib.rs`.

I don't think the `fruit` example really represent a real-world, and I'm not sure the `animation` example really represents a good way to do animation in production use (though it didn't perform well when I tried to change it to use no-copy drawing without pre-rendering).